placehero/src/lib.rs

```diff
@@ -84,21 +91,41 @@ pub fn run(event_loop: EventLoopBuilder) -> Result<(), EventLoopError> {
 impl Placehero {
     fn sidebar(&mut self) -> impl WidgetView<Self> + use<> {
         if let Some(instance) = &self.instance {
+            let back = if self.show_context.is_some() {
+                // TODO: Make the ⬅️ arrow not be available to screen readers.
+                Some(button("⬅️ Back to Timeline", |app_state: &mut Self| {
+                    // TODO: Maintain scroll state in the timeline.
+                    // TODO: More of a back stack.
+                    app_state.show_context = None;
+                    app_state.context = None;
+                }))
+            } else {
+                None
+            };
             Either::A(flex((
                 label("Connected to:"),
                 // TODO: We should probably use an ArcStr for this?
                 prose(instance.title.as_str()),
+                back,
             )))
         } else {
             Either::B(prose("Not yet connected (or other unhandled error)"))
         }
     }
 
     fn main_view(&mut self) -> impl WidgetView<Self> + use<> {
-        if self.statuses.is_empty() {
-            Either::A(prose("No statuses yet loaded"))
+        if let Some(show_context) = self.show_context.as_ref() {
+            if let Some(context) = self.context.as_ref() {
+                // TODO: Display the status until the entire thread loads; this is hard because
+                // the thread's scroll position would jump.
```

Comment:
Would using the virtual scroll widget help with that scenario?

Reply:
I think that's right, yes.